exercise/templatetags/exercise.py

```diff
@@ -118,6 +118,81 @@ def submission_status(status):
     return Submission.STATUS[status]
 
 
+"""
+    Used for adding relevant classes to the
+    points_badge to enable textual status.
+"""
+@register.simple_tag(takes_context=True)
+def get_points_class(context):
+    html_class = set()
+    confirm_the_level = context['confirm_the_level']
+    if context['submitted'] and context['official']:
+
+        # confirm_the_level requires also that graded = True
+        if ((confirm_the_level and context['graded'])
+        or not confirm_the_level):
+            if context['full_score']:
+                html_class.add("badge-success")
+
+            elif (context['passed'] or
+            (confirm_the_level and context['confirmable_points'])):
+                html_class.add("badge-warning")
+            else:
+                html_class.add("badge-danger")
+
+            # Unconfirmed is not handled if confirm_the_level == True
+            if (not confirm_the_level and 'unconfirmed' in context
+            and context['unconfirmed']):
+                html_class.add("unconfirmed-points")
+
+
+    # Handles the special difficulty "-".
+    if context['difficulty'] == "-" and context['max'] <= 2:
+        html_class.add("aplus-points")
+        class_name = ""
+
+        if ((context['max'] == 2 and context['points'] == 2)
+        or context['max'] == 1 and context['points']== 1):
+            class_name = "accepted"
+        elif context['max'] == 2 and context['points'] == 1:
+            class_name = "ok"
+        elif context['points'] == 0 and context['submitted']:
+            class_name = "rejected"
+
+        html_class.add(class_name)
+    return " ".join(html_class)
+
+
+#Adds the content of the points_badge.
+@register.simple_tag(takes_context=True)
+def get_points_text(context):
+    points_string = str(context['points']) + " / " + str(context['max'])
+    if context['confirm_the_level']:
+        if context['submitted']:
+            if not context['official']:
+                points_string = _("Submitted after deadline")
+            elif context['passed']:
+                points_string = _("Accepted")
+            elif context['graded']:
+                points_string = _("Rejected")
+            else:
+                points_string = _("Submitted")
+        elif context.get('confirmable_points'):
+            points_string = _("Missing")
+        elif context['exercise_page']:
+            points_string = _("Not submitted")
+
+    elif context['difficulty'] == "-" and context['max'] <= 2:
```

Comment:
If we change this to: `if (context['confirm_the_level'] or context['difficulty'] == "-") and context['max'] <= 2:` and add one block on below list below

```
elif context.get('confirmable_points');:
  points_string = _("Missing")
elif context['confirm_the_level'] and :context['exercise_page']
  points_string = _("Not submitted")
```

Then these codes should yield the same result for both difficulty ' -' and confirm_the_level cases. Submitted after deadline will be dropped, but that doesn't seem any way logical as it does remove the information about the grade (doesn't tell if the submission was accepted or not).